### ISSUE TITLE
RNaD: Possible Error in calculation of Neurd Loss

### ISSUE THREAD

**Issue by spktrm** [google-deepmind/open_spiel#1156]
In this line of the RNaD algorithm
https://github.com/google-deepmind/open_spiel/blob/7c58b6c28cc975f2a7a02a7ea604197a30ed3186/open_spiel/python/algorithms/rnad/rnad.py#L574

Should the line instead be this? This is so we only subtract the mean calculated from the valid logits.
```python
logits = logit_pi - (jnp.sum(
        logit_pi * legal_actions, axis=-1, keepdims=True) / jnp.sum(legal_actions, axis=-1, keepdims=True))
```

As a result, should the line below be an average over actions rather than a sum?
https://github.com/google-deepmind/open_spiel/blob/7c58b6c28cc975f2a7a02a7ea604197a30ed3186/open_spiel/python/algorithms/rnad/rnad.py#L579
i.e.
```python
nerd_loss = jnp.sum(
        legal_actions *
        apply_force_with_threshold(logits, adv_pi, threshold, threshold_center),
        axis=-1) / jnp.sum(legal_actions, axis=-1)
```

This is particularly relevant in games where there is frequently a number of invalid actions.

**Comment by lanctot** [comment]
@perolat can you take a look?

**Comment by lanctot** [comment]
Hi @spktrm , I spoke to Julien.

He said you're correct about the first one, can you submit a PR? 

The second one could go either way: it's just a matter of knowing what works. It is not clear whether one works better than the other and it might end up being similar behavior but require different hyper-parameters. Maybe you can try it and let us know?

**Comment by spktrm** [comment]
I have submitted a PR regarding the first point here: https://github.com/google-deepmind/open_spiel/pull/1157, thank you for the opportunity to contribute :).

With regards to the second point, I will experiment further with the fix I am suggesting and let you know how it goes.

Meanwhile, is it possible to provide clarity on these other issues? Namely:
- https://github.com/google-deepmind/open_spiel/issues/1109
- https://github.com/google-deepmind/open_spiel/issues/1075

**Comment by lanctot** [comment]
Hi @spktrm,

Yeah I will make Julien aware of those (sorry, I thought they were resolved already). 

I think it may be useful to also try contacting him directly by email, though... because I'm mostly just relaying messages from here to him and back :)

**Comment by spktrm** [comment]
Thank you. What is his best email?

**Comment by lanctot** [comment]
> Thank you. What is his best email?

Still the same one from the Mastering Stratego paper.

**Comment by lanctot** [comment]
Fixed by https://github.com/google-deepmind/open_spiel/pull/1157, which has now been merged into master.
